src/AggregateFunctions/AggregateFunctionGroupSortedArray.h

```diff
@@ -0,0 +1,182 @@
+#pragma once
+
+#include <Columns/ColumnArray.h>
+#include <DataTypes/DataTypeArray.h>
+
+#include <AggregateFunctions/IAggregateFunction.h>
+#include "AggregateFunctionGroupSortedArrayData.h"
+
+namespace DB
+{
+template <typename TT, bool is_plain_column>
+inline TT readItem(const IColumn * column, Arena * arena, size_t row)
+{
+    if constexpr (std::is_same_v<TT, StringRef>)
+    {
+        if constexpr (is_plain_column)
+        {
+            StringRef str = column->getDataAt(row);
+            auto ptr = arena->alloc(str.size);
+            std::copy(str.data, str.data + str.size, ptr);
+            return StringRef(ptr, str.size);
+        }
+        else
+        {
+            const char * begin = nullptr;
+            return column->serializeValueIntoArena(row, *arena, begin);
+        }
+    }
+    else
+    {
+        return column->getUInt(row);
```

Comment:
Why is this reading a UInt and not a type based on the column type? Maybe I'm missunderstanding the function but I understand that is something like:
FUNC(LIMIT)(COLUMN1, [COLUMN2])
that returns an array topN elements of Column1, optionally ordered by Column2 and otherwise ordered by themselves (if they are comparable)
-> LIMIT: Numbers of element to keep
-> COLUMN1: Values to keep. This can be of any type (at least any type that can be held in an array, which I guess is everything)
-> COLUMN2: Order of each element. This can only be of type Int64.

Reply:
I took it from topk(), they only use UInt64 for second parameter, I will review if I can use with any numeric value here.

Reply:
Your interpretation is correct, I have some incoherences with this UInt64/Int64.
The problem here, as far as I can see, is the comparison with negative values, apparently topk() doesn't support it but I understand we can get lowest values using **group...(a, b)** or highest using **group....(a,-b)** and then it requires supporting signed. But at the same time I am always storing UInt64, this is not correct.